File: src/test/java/com/winterhate/strategy/handlers/HandlerFirst.java
package com.winterhate.strategy.handlers;

import lombok.NonNull;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class HandlerFirst extends HandlerStrategy {
    @Override
    public @NonNull Handlers handles() {
        return Handlers.FIRST;
    }

    @Override
    public void apply(HandlerDto handlerDto) {
        log.info("Handled by HandlerFirst: {}", handlerDto);
    }
}

File: src/test/kotlin/com/winterhate/strategy/handlers/HandlerFirst.kt
package com.winterhate.strategy.handlers

import org.slf4j.Logger
import org.springframework.stereotype.Component

@Component
class HandlerFirst(private val log: Logger) : HandlerStrategy() {

    override fun handles(): Handlers = Handlers.FIRST

    override fun apply(data: HandlerDto) {
        log.info("Handled by HandlerFirst: {}", data)
    }

}

